src/main/scala/swiss/dasch/domain/AssetModel.scala

```diff
@@ -131,46 +131,45 @@ object OriginalFile {
 }
 
 sealed trait DerivativeFile(file: Path) {
+  def assetId: AssetId
   final def toPath: Path             = file
   final def filename: NonEmptyString = NonEmptyString.unsafeFrom(file.filename.toString)
-  final def assetId: AssetId         = AssetId.fromPath(file).head
 }
 
 object DerivativeFile {
-  final case class JpxDerivativeFile private (file: Path) extends DerivativeFile(file)
+  private def from[A <: DerivativeFile](
+    file: Path,
+    isSupported: String => Boolean,
+    f: (Path, AssetId) => A
+  ): Either[String, A] =
+    file match {
+      case hidden if hidden.filename.toString.startsWith(".") => Left("Hidden file")
+      case derivative if isSupported(derivative.fileExtension) =>
+        AssetId.fromPath(derivative).map(f(file, _)).toRight("No AssetId in filename")
+      case _ => Left("Unsupported file type")
```

Comment:
```suggestion
      case hidden if hidden.filename.toString.startsWith(".") => Left("Hidden file.")
      case derivative if isSupported(derivative.fileExtension) =>
        AssetId.fromPath(derivative).map(f(file, _)).toRight("No Asset ID in the filename.")
      case _ => Left("Unsupported file type.")
```

To keep it unified across all `AssetId` related messages.